vendor/engines/secure_rooms/spec/models/user_spec.rb

```diff
@@ -3,6 +3,36 @@
 require "rails_helper"
 
 RSpec.describe User do
-  subject(:user) { build(:user) }
+  subject(:user) { build(:user, :netid) }
```

Comment:
You can DRY things up a bit here by setting card_number in the subject call:
```suggestion
  subject(:user) { build(:user, :netid, card_number: card_number) }
```

Then in each context block, you just need to define `card_number`:
`let(:card_number) { "12345-123" }`
... and you can drop the `before` blocks (`before { user.update(card_number: card_number) }`)

Reply:
We tried this but it didn't seem to DRY up the code much since we're changing the card_number so much and the tests started failing.

Reply:
I pushed up a commit that does this.  It doesn't DRY things up much, but the intent is a little clearer (for me at least).  Calling `user.update` was actually persisting the record to the db, which makes the specs pass.  I refactored a bit so the user under test is instantiated up front before we look it up by the card number.